Client/mods/deathmatch/logic/CClientEntity.cpp

```diff
@@ -458,33 +458,29 @@ bool CClientEntity::GetCustomDataBool(const char* szName, bool& bOut, bool bInhe
     return false;
 }
 
-void CClientEntity::SetCustomData(const char* szName, const CLuaArgument& Variable, bool bSynchronized)
+bool CClientEntity::SetCustomData(const char* szName, const CLuaArgument& Variable, bool bSynchronized)
 {
-    assert(szName);
     if (strlen(szName) > MAX_CUSTOMDATA_NAME_LENGTH)
     {
         // Don't allow it to be set if the name is too long
         CLogger::ErrorPrintf("Custom data name too long (%s)", *SStringX(szName).Left(MAX_CUSTOMDATA_NAME_LENGTH + 1));
-        return;
+        return false;
     }
 
     // Grab the old variable
-    CLuaArgument oldVariable;
-    SCustomData* pData = m_pCustomData->Get(szName);
-    if (pData)
+    CLuaArgument OldVariable;
+    if (m_pCustomData->Set(szName, Variable, bSynchronized, &OldVariable))
```

Comment:
Does it work correctly if there is no previous variable?

Reply:
It does.